Src/GoogleApis.Auth.DotNet4.Tests/OAuth2/DefaultCredentialProvider/DefaultCredentialProviderTests.cs

```diff
@@ -0,0 +1,232 @@
+
+            var credential = credentialProvider.GetApplicationDefaultCredential();
+
+            Assert.IsInstanceOf(typeof(ServiceAccountCredential), credential.UnderlyingCredential);
+            Assert.IsTrue(credential.IsCreateScopedRequired);
+
+            var scopes = new[] { "https://www.googleapis.com/auth/cloud-platform" };
+            var scopedCredential = credential.CreateScoped(scopes);
+            Assert.IsTrue(credential != scopedCredential);
+
+            Assert.IsFalse(scopedCredential.IsCreateScopedRequired);
+            CollectionAssert.AreEqual(scopes, ((ServiceAccountCredential) scopedCredential.UnderlyingCredential).Scopes);
+        }
+
+        #endregion
+
+        #region Invalid Cases
+
+        /// <summary> No credential files or environment variable specified - like a fresh developer's machine.</summary>
+        [Test]
+        public void GetApplicationDefaultCredentials_NoCredentialFiles()
+        { 
+            try
+            {
+                var credential = credentialProvider.GetApplicationDefaultCredential();
+                Assert.Fail();
+            }
+            catch (InvalidOperationException e)
+            {
+                Assert.IsTrue(e.Message.Contains("The Application Default Credentials are not available"));
+            }
+        }
+
+        /// <summary>Environment variable points to a non existant credential file.</summary>
+        [Test]
+        public void GetApplicationDefaultCredentials_MissingCredentialFile()
+        {
+            credentialProvider.SetEnvironmentVariable(CredentialEnvironmentVariable, WellKnownCredentialFilePath);
+            
+            try
+            {
+                var credential = credentialProvider.GetApplicationDefaultCredential();
+                Assert.Fail();
+            }
+            catch (InvalidOperationException e)
+            {
+                Assert.IsTrue(e.Message.Contains("Please check the value of the Environment Variable GOOGLE_APPLICATION_CREDENTIALS"));
+            }
+        }
+
+        /// <summary>Credential file has invalid content format.</summary>
+        [Test]
+        public void GetApplicationDefaultCredentials_InvalidCredentialFile()
+        {
+            string credentialFilepath = "TempFilePath.json";
+            credentialProvider.SetEnvironmentVariable(CredentialEnvironmentVariable, credentialFilepath);
+            credentialProvider.SetFileContents(credentialFilepath, "Invalid Credentials File Contents");
+
+            try
+            {
+                var credential = credentialProvider.GetApplicationDefaultCredential();
+                Assert.Fail();
+            }
+            catch (InvalidOperationException e)
+            {
+                Assert.IsTrue(e.Message.Contains("Error reading credential file from location"));
+            }
+        }
+
+        // TODO(jtattermusch): test compute engine credential.
+
+        // TODO(jtattermusch): add more tests
```

Comment:
This was basically a note to self. I added a few more tests and removed the TODO.